pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -744,7 +746,11 @@ private static void applyAlias(Map<Identifier, Expression> aliasMap, Expression
 
   private static List<String> extractOptionsFromSql(String sql) {
     List<String> results = new ArrayList<>();
-    Matcher matcher = OPTIONS_REGEX_PATTEN.matcher(sql);
+    Matcher matcher = COMMENTED_QUERY_PATTERN.matcher(sql);
+    if (matcher.find()) {
```

Comment:
This will ignore all the options if any line of the sql is commented out.

We should only ignore the options within the commented out part, so the options should not be ignored in this query:
`select ... option(...) -- commented out`

Some other corner cases not handled are (not necessary handle in this PR, but might worth add a TODO):
`option()` within literal or identifier

Reply:
Makes sense, fixing this

Reply:
@Jackie-Jiang have incorporated this fix in PR: https://github.com/apache/pinot/pull/7894